### PR TITLE
Maven multi module skeleton

### LINKED ISSUE #10: Create initial maven modules
### Description
As a team member, I need to have the initial skeleton of the project created so I can proceed with implementation.

### Requirements

- create modules according to https://github.com/ita-social-projects/Home/wiki/Multi-Module-structure
- all the modules should have their own pom.xml file
- at this point leave poms as plain as it can be

### Acceptance
- [ ] All the modules are created
- [ ] Maven build runs without issues

### Epic
 #3

### PR DESCRIPTION
**Pull request by isvyry**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [x] @alexorlenko
- [ ] @Arthurpanda
- [ ] @shiftko  
- [x] @mykytam 
- [x] @omelchenkotimur 
- [x] @Prosperro-de  

### Second Level Review

- [ ] @DzigMS 
- [ ] @kh0ma 

## Summary of issue

This PR resolves issue #10 

## Summary of change

Create multi-module project skeleton

## Testing approach

-

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: home-service/pom.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    </parent>
+    <packaging>jar</packaging>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>home-service</artifactId>
+    <dependencies>
+        <dependency>
+            <artifactId>home-data</artifactId>
+            <groupId>com.project</groupId>
+            <version>0.0.1-SNAPSHOT</version>
```
Comment (yevhenmanaienkov) [inline home-service/pom.xml]: Move versions into a properties variable.

File: home-api-tests/pom.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        <artifactId>home</artifactId>
+        <groupId>com.project</groupId>
+        <version>0.0.1-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <packaging>jar</packaging>
+    <artifactId>home-api-tests</artifactId>
+
+</project>
```
Comment (kh0ma) [inline home-api-tests/pom.xml]: No newline at the end of the file

Comment (kh0ma) [reply home-api-tests/pom.xml]: https://stackoverflow.com/a/5813359

File: home-api-tests/pom.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>home</artifactId>
+        <groupId>com.project</groupId>
+        <version>0.0.1-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <packaging>jar</packaging>
```
Comment (kh0ma) [inline home-api-tests/pom.xml]: Why do we need a packing jar for tests?

File: home-clients/pom.xml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>home</artifactId>
+        <groupId>com.project</groupId>
+        <version>0.0.1-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <packaging>jar</packaging>
```
Comment (kh0ma) [inline home-clients/pom.xml]: We are not going to pack this module to jar. Generated artifacts should be installed instead.

File: home-data/pom.xml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        <version>0.0.1-SNAPSHOT</version>
+    </parent>
+    <packaging>jar</packaging>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>home-data</artifactId>
+    <dependencies>
+        <dependency>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-starter-data-jpa</artifactId>
```
Comment (kh0ma) [inline home-data/pom.xml]: indentation

File: home-open-api/pom.xml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>home</artifactId>
+        <groupId>com.project</groupId>
```
Comment (kh0ma) [inline home-open-api/pom.xml]: I think we need to find out our group id for the whole project. CC: @DzigMS

File: home-server-stub/pom.xml [hunk 7]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
```
Comment (kh0ma) [inline home-server-stub/pom.xml:2]: This module will be generated for us. I don't think we need a module for it but we need a module that will generate it.

File: pom.xml [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+        <module>home-data</module>
+        <module>home-service</module>
+        <module>home-application</module>
+        <module>home-clients</module>
+        <module>home-open-api</module>
+        <module>home-api-tests</module>
+        <module>home-server-stub</module>
+        <module>home-data-migration</module>
+    </modules>
+    <parent>
```
Comment (kh0ma) [inline pom.xml]: Why did we place it here?

File: pom.xml [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+    <artifactId>home</artifactId>
+    <version>0.0.1-SNAPSHOT</version>
+    <name>home</name>
+    <description>SoftServe Home Project</description>
+
+    <packaging>pom</packaging>
+
+    <properties>
+        <java.version>11</java.version>
+        <start-class>com.project.HomeApplication</start-class>
```
Comment (kh0ma) [inline pom.xml]: IMO this should be in app module.

File: home-open-api/pom.xml (hunk 6 above)
Comment (kh0ma) [reply home-open-api/pom.xml]: `com.softserveinc.ita.homeproject`

**Review by alexorlenko — APPROVED** [review]

**Review by Arthurpanda — APPROVED** [review]

**Review by Prosperro-de — APPROVED** [review]

**Review by mykytam — APPROVED** [review]
Igor did a great job!

**Pull request merged**